### PR TITLE
sgr 0.3.12 (new formula)

### PR DESCRIPTION
**Pull request by onpaws**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

[sgr](https://www.splitgraph.com/docs/sgr-advanced/getting-started/introduction) is an open source tool for manipulating and publishing [Splitgraph](https://www.splitgraph.com/docs/sgr-advanced/getting-started/frequently-asked-questions) data images, written in Python.

The [tap](https://github.com/splitgraph/homebrew-splitgraph) is available and can be installed
```
brew install splitgraph/splitgraph/sgr
```

### CODE REVIEW

**Comment by onpaws** [comment]
Results of `brew test`:
```
$ brew test --verbose sgr
==> Testing sgr
==> /usr/local/Cellar/sgr/0.3.12/bin/sgr cloud login --username homebrewtest --password correcthorsebattery 2>&1
```

**Comment by onpaws** [comment]
Looks like everything passed 🎉 

image

File: Formula/sgr.rb [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+  resource "splitgraph-pipelinewise-target-postgres" do
+    url "https://files.pythonhosted.org/packages/59/54/de6a8a2b6bdb24de8d8fd4a2465532f3523abc750af4dd9d6e5c17ce6a53/splitgraph-pipelinewise-target-postgres-2.1.0.tar.gz"
+    sha256 "9d100ac65288ce24a90da159bbbb06f0fdc0871c2815c63bb6417fea7df4894f"
+  end
+
+  resource "tabulate" do
+    url "https://files.pythonhosted.org/packages/7a/53/afac341569b3fd558bf2b5428e925e2eb8753ad9627c1f9188104c6e0c4a/tabulate-0.8.10.tar.gz"
+    sha256 "6c57f3f3dd7ac2782770155f3adb2db0b1a269637e42f27599925e64b114f519"
+  end
```
Comment (carlocab) [inline Formula/sgr.rb:163]: Can we change this with `depends_on "libpython-tabulate"`?

File: Formula/sgr.rb [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    url "https://files.pythonhosted.org/packages/0e/e7/e705ead133d21de4be752af4b3a0cb1f02514ff45bf165b3955c1ce22077/websocket-client-1.3.3.tar.gz"
+    sha256 "d58c5f284d6a9bf8379dab423259fe8f85b70d5fa5d2916d5791a84594b122b1"
+  end
+
+  def install
+    venv = virtualenv_create(libexec, "python3")
+    venv.pip_install resources
+    poetry = Formula["poetry"].opt_bin/"poetry"
+    system poetry, "build", "--format", "wheel", "--verbose", "--no-interaction"
+    venv.pip_install_and_link Dir["dist/splitgraph-*.whl"].first
```
Comment (carlocab) [inline Formula/sgr.rb:190]:
```suggestion
    venv.pip_install_and_link buildpath.glob("dist/splitgraph-*.whl").first
```
We're not operating on directories, so let's avoid `Dir`. (Yes other formulae that use `Dir` here probably should be fixed too)

File: Formula/sgr.rb [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      Docker is available and follow the install steps [3].
+
+      Example projects https://github.com/splitgraph/sgr/tree/master/examples
+      Five minute demo https://www.splitgraph.com/docs/sgr-advanced/getting-started/five-minute-demo
+
+      [1] https://www.splitgraph.com/docs/sgr-advanced/architecture/splitgraph-engine
+      [2] https://hub.docker.com/r/splitgraph/engine
+      [3] https://www.splitgraph.com/docs/sgr-advanced/getting-started/installation#manual
+    EOS
+  end
```
Comment (carlocab) [inline Formula/sgr.rb:207]:
```suggestion
```
We only accept `caveats` that are specific to Homebrew.

**Review by carlocab — COMMENTED** [review]
Thanks for your PR. A few comments.

**Comment by onpaws** [comment]
Thank you @carlocab !
I'm reviewing your comments now and will post back

File: Formula/sgr.rb (hunk 2 above)
Comment (onpaws) [reply Formula/sgr.rb:190]: I indeed saw the `Dir` approach in other formulae. Understood, changed.

File: Formula/sgr.rb (hunk 3 above)
Comment (onpaws) [reply Formula/sgr.rb:207]: Understood, dropped.

File: Formula/sgr.rb (hunk 1 above)
Comment (onpaws) [reply Formula/sgr.rb:163]: Changed.

I used `brew audit --new-formula --strict sgr` to check and no errors. 
I'm additionally checking by `brew install`-ing it now. Will post back

Comment (onpaws) [reply Formula/sgr.rb]: Looking good!
```
==> /usr/local/Cellar/sgr/0.3.12/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/sgr-20220729-87128-6ul5ev/sgr-0.3.12/dist/splitgraph-0.3.12-py3-none-any.whl
Using pip 22.1.2 from /usr/local/Cellar/sgr/0.3.12/libexec/lib/python3.10/site-packages/pip (python 3.10)
Processing ./dist/splitgraph-0.3.12-py3-none-any.whl
Installing collected packages: splitgraph
  changing mode of /usr/local/Cellar/sgr/0.3.12/libexec/bin/sgr to 755
Successfully installed splitgraph-0.3.12

[notice] A new release of pip available: 22.1.2 -> 22.2.1
[notice] To update, run: python3.10 -m pip install --upgrade pip
==> Cleaning
==> Finishing up
ln -s ../Cellar/sgr/0.3.12/bin/sgr sgr
==> Summary
🍺  /usr/local/Cellar/sgr/0.3.12: 3,556 files, 41.8MB, built in 45 minutes 11 seconds
==> Running `brew cleanup sgr`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```

Thanks for the tip, TIL tabulate is available via `depends_on`

**Comment by onpaws** [comment]
I made the suggested changes and they seem to be working on my machine nicely. Thanks for the tips! (

I did a `git commit --amend` + force push. Let's see what test-bot thinks 🍿

**Comment by onpaws** [comment]
> I did a `git commit --amend` + force push. Let's see what test-bot thinks 🍿

Looks like test-bot is happy 🎉

File: Formula/sgr.rb [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    sha256 "d58c5f284d6a9bf8379dab423259fe8f85b70d5fa5d2916d5791a84594b122b1"
+  end
+
+  def install
+    venv = virtualenv_create(libexec, "python3")
+    venv.pip_install resources
+    poetry = Formula["poetry"].opt_bin/"poetry"
+    system poetry, "build", "--format", "wheel", "--verbose", "--no-interaction"
+    venv.pip_install_and_link buildpath.glob("dist/splitgraph-*.whl").first
+    bin.install_symlink libexec/"bin/sgr"
```
Comment (carlocab) [inline Formula/sgr.rb:187]: Doesn't this symlink get created by `#pip_install_and_link`?

**Review by carlocab — COMMENTED** [review]
Thanks; one last question and I'm happy with this.

File: Formula/sgr.rb (hunk 4 above)
Comment (onpaws) [reply Formula/sgr.rb:187]: I did [try](https://gist.github.com/onpaws/4a4ae8e48d84756d02ed268e82a925d8#file-brew-sh-L5547) `#pip_install_and_link` by itself first, but I observed the symlink did [not](https://gist.github.com/onpaws/4a4ae8e48d84756d02ed268e82a925d8#file-trying-to-run-it-sh) get created.

Thanks for your diligent review, it's appreciated!

**Review by carlocab — APPROVED** [review]
Great, thanks for your work!

**Review by alebcay — APPROVED** [review]

**Pull request closed without merging**